components/ai_chat/core/browser/associated_content_manager_unittest.cc

```diff
@@ -0,0 +1,292 @@
+  EXPECT_EQ("test-turn-uuid",
+            associated_content[1]->conversation_turn_uuid.value());
+
+  // UUIDs should be match
+  EXPECT_EQ(first_content.uuid(), associated_content[0]->uuid);
+  EXPECT_EQ(second_content.uuid(), associated_content[1]->uuid);
+
+  // GetCachedContentsMap should work and have both content items under the same
+  // turn UUID
+  auto contents_map = conversation_handler_->associated_content_manager()
+                          ->GetCachedContentsMap();
+  EXPECT_TRUE(contents_map.contains("test-turn-uuid"));
+  EXPECT_EQ(2u, contents_map.at("test-turn-uuid").size());
+}
+
+TEST_F(AssociatedContentManagerUnitTest,
+       AssociateUnsentContentWithTurn_MultipleContent_MultipleTurns) {
+  NiceMock<MockAssociatedContent> content;
+  conversation_handler_->associated_content_manager()->AddContent(&content);
+
+  auto turn1 = mojom::ConversationTurn::New(
+      "turn-1", mojom::CharacterType::HUMAN, mojom::ActionType::QUERY,
+      "Test human message 1", std::nullopt, std::nullopt, std::nullopt,
+      base::Time::Now(), std::nullopt, std::nullopt, false, std::nullopt);
+
+  auto turn2 = mojom::ConversationTurn::New(
+      "turn-2", mojom::CharacterType::HUMAN, mojom::ActionType::QUERY,
+      "Test human message 2", std::nullopt, std::nullopt, std::nullopt,
+      base::Time::Now(), std::nullopt, std::nullopt, false, std::nullopt);
+
+  // Associate the first content with turn 1
+  conversation_handler_->associated_content_manager()
+      ->AssociateUnsentContentWithTurn(turn1);
+
+  auto associated_content = conversation_handler_->associated_content_manager()
+                                ->GetAssociatedContent();
+  ASSERT_EQ(1u, associated_content.size());
+  EXPECT_EQ("turn-1", associated_content[0]->conversation_turn_uuid);
+
+  // Add a second content delegate
+  auto second_content = std::make_unique<NiceMock<MockAssociatedContent>>();
+  conversation_handler_->associated_content_manager()->AddContent(
+      second_content.get());
+  conversation_handler_->associated_content_manager()
+      ->AssociateUnsentContentWithTurn(turn2);
+
+  // Both content items should be associated with the turn
+  associated_content = conversation_handler_->associated_content_manager()
+                           ->GetAssociatedContent();
+  ASSERT_EQ(2u, associated_content.size());
+
+  // Both should have the same conversation_turn_uuid
+  EXPECT_EQ("turn-1", associated_content[0]->conversation_turn_uuid);
+  EXPECT_EQ("turn-2", associated_content[1]->conversation_turn_uuid);
+
+  // GetCachedContentsMap should work and have both content items under their
+  // associated turns.
+  auto contents_map = conversation_handler_->associated_content_manager()
+                          ->GetCachedContentsMap();
+  EXPECT_TRUE(base::Contains(contents_map, "turn-1"));
+  EXPECT_EQ(1u, contents_map.at("turn-1").size());
+
+  EXPECT_TRUE(base::Contains(contents_map, "turn-2"));
+  EXPECT_EQ(1u, contents_map.at("turn-2").size());
```

Comment:
This only checks the key exists but we also need to check value are exactly the content we associate